app/lib/client_sorter.rb

```diff
@@ -36,8 +37,10 @@ def filtered_clients
               else
                 @clients.after_consent
               end
-    # Force an inner join to `intakes` to exclude clients from previous years
-    clients = clients.joins(:intake)
+    # Filter on product_year to only show clients who used this-year's product
+    if @use_product_year
+      clients = clients.where(intake: Intake.where(type: "Intake::CtcIntake", product_year: MultiTenantService.new(:ctc).current_product_year).or(Intake.where(type: "Intake::GyrIntake", product_year: MultiTenantService.new(:gyr).current_product_year)))
```

Comment:
remove type, just look for product year (should there be a rails.configuration.product_year ?)

Reply:
also this should use `filterable_product_year` from client which should be the same value as intakes.product_year